aten/src/ATen/core/Macros.h

```diff
@@ -3,41 +3,7 @@
 #include <sstream>
 #include <string>
 
-// You can use the definition AT_CORE_STATIC_WINDOWS to control whether
-// or not we apply __declspec.  You will want to set this as
-// -DAT_CORE_STATIC_WINDOWS=1 when compiling code which links
-// against ATen/core on Windows, when ATen/core is built as a
-// static library (in which case, saying the symbol is coming
-// from a DLL would be incorrect).
-
-#ifdef _WIN32
-#if !defined(AT_CORE_STATIC_WINDOWS)
-#define AT_CORE_EXPORT __declspec(dllexport)
-#define AT_CORE_IMPORT __declspec(dllimport)
-#else // !defined(AT_CORE_STATIC_WINDOWS)
-#define AT_CORE_EXPORT
-#define AT_CORE_IMPORT
-#endif // !defined(AT_CORE_STATIC_WINDOWS)
-#else  // _WIN32
-#if defined(__GNUC__)
-#define AT_CORE_EXPORT __attribute__((__visibility__("default")))
-#else // defined(__GNUC__)
-#define AT_CORE_EXPORT
-#endif // defined(__GNUC__)
-#define AT_CORE_IMPORT AT_CORE_EXPORT
-#endif  // _WIN32
-
-// AT_CORE_API is a macro that, depends on whether you are building the
-// main library or not, resolves to either AT_CORE_EXPORT or
-// AT_CORE_IMPORT.
-//
-
-// TODO: unify the controlling macros.
-#if defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
-#define AT_CORE_API AT_CORE_EXPORT
-#else // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
-#define AT_CORE_API AT_CORE_IMPORT
-#endif // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
+#include "c10/macros/Macros.h"
```

Comment:
What's the rule for subfolders under c10? My preference (evidenced by `ATen/core/` structure) is that most headers live in the top level.

Reply:
At some point we should just include a unified c10/c10.h that includes all dependent headers, and have c10 manage the internal header relationships - does that make sense? I think eigen uses this as well by including headers like <Eigen/Core> without exposing further details.